tests/llvm/test_helpers.py

```diff
@@ -519,6 +519,48 @@ def test_helper_is_boolean(self, mode, ir_type, expected):
 
         assert res == expected
 
+    @pytest.mark.llvm
+    @pytest.mark.parametrize('mode', ['CPU',
+                                      pytest.param('PTX', marks=pytest.mark.cuda)])
+    @pytest.mark.parametrize('ir_type,expected', [
+        (DOUBLE_VECTOR_TYPE, (1,)),
+        (DOUBLE_VECTOR_PTR_TYPE, (1,)),
+        (DOUBLE_MATRIX_TYPE, (1,1)),
+        (DOUBLE_MATRIX_PTR_TYPE, (1,1)),
+    ], ids=str)
+    def test_helper_get_array_shape(self, mode, ir_type, expected):
+        with pnlvm.LLVMBuilderContext() as ctx:
+            func_ty = ir.FunctionType(ir.VoidType(), [ir.IntType(32).as_pointer()])
+
+            custom_name = ctx.get_unique_name("is_boolean")
+            function = ir.Function(ctx.module, func_ty, name=custom_name)
+            out = function.args[0]
+            block = function.append_basic_block(name="entry")
+            builder = ir.IRBuilder(block)
+
+            variable = builder.load(builder.alloca(ir_type))
+            assert pnlvm.helpers.get_array_shape(variable) == expected
```

Comment:
this should need compilation at all.
`assert pnlvm.helpers.get_array_shape(ir_type(None)) == expected` should do fine